kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -30,59 +37,212 @@ pub(crate) fn goto_definition(
+/// Similar to vars.rs/reesolver_var, find a ScopeObj corresponding to the definition of identifier
+pub(crate) fn resolve_var(
+    names: &[String],
+    pkgpath: &str,
+    current_scope: &Scope,
+    scope_map: &IndexMap<String, Rc<RefCell<Scope>>>,
+) -> Option<ScopeObject> {
+    match names.len() {
+        0 => None,
+        1 => {
+            let name = names[0].clone();
+            match current_scope.lookup(&name) {
+                Some(obj) => Some(obj.borrow().clone()),
+                None => None,
+            }
+        }
+        _ => {
+            if pkgpath.is_empty() {
+                let name = names[0].clone();
+                match current_scope.lookup(&name) {
+                    Some(obj) => {
+                        match &obj.borrow().ty.kind {
+                            kclvm_sema::ty::TypeKind::Schema(schema_type) => {
+                                find_attr_in_schema(
+                                    schema_type,
+                                    &names[1..],
+                                    pkgpath,
+                                    // current_scope,
+                                    scope_map,
+                                )
+                            }
+                            kclvm_sema::ty::TypeKind::Module(_) => {
+                                let pkg = name;
+                                match scope_map.get(&pkg) {
+                                    Some(scope) => {
+                                        return resolve_var(
+                                            &names[1..],
+                                            pkgpath,
+                                            &scope.borrow(),
+                                            scope_map,
+                                        )
+                                    }
+                                    None => None,
+                                }
+                            }
+                            _ => None,
+                        }
+                    }
+                    None => {
+                        let pkg = name;
+                        match scope_map.get(&pkg) {
+                            Some(scope) => {
+                                resolve_var(&names[1..], pkgpath, &scope.borrow(), scope_map)
+                            }
+                            None => None,
+                        }
+                    }
+                }
+            } else {
+                match scope_map.get(pkgpath) {
+                    Some(scope) => {
+                        return resolve_var(&names[1..], pkgpath, &scope.borrow(), scope_map)
+                    }
+                    None => None,
+                }
+            }
+        }
     }
+}
 
-    res
+pub fn find_attr_in_schema(
+    schema_type: &SchemaType,
+    names: &[String],
+    pkgpath: &str,
+    // current_scope: &Scope,
+    scope_map: &IndexMap<String, Rc<RefCell<Scope>>>,
```

Comment:
Use a context named just like `SemanticModel` to avoid fragmented transmission of scope parameters